equinix/resource_fabric_service_profile.go

```diff
@@ -880,49 +880,42 @@ func customFieldFabricSpToTerra(customFieldl []v4.CustomField) []map[string]inte
 	return mappedCustomFieldl
 }
 
-func processStepFabricSpToTerra(processStepl []v4.ProcessStep) []map[string]interface{} {
+func processStepFabricSpToTerra(processStepl []v4.ProcessStep) []interface{} {
 	if processStepl == nil {
 		return nil
 	}
-	mappedProcessStepl := make([]map[string]interface{}, len(processStepl))
-	for index, processStep := range processStepl {
-		mappedProcessStepl[index] = map[string]interface{}{
-			"title":       processStep.Title,
-			"sub_title":   processStep.SubTitle,
-			"description": processStep.Description,
-		}
+	mappedProcessStep := map[string]interface{}{
+		"title":       processStepl[0].Title,
+		"sub_title":   processStepl[0].SubTitle,
+		"description": processStepl[0].Description,
 	}
-	return mappedProcessStepl
+
+	return []interface{}{mappedProcessStep}
 }
 
 func marketingInfoMappingToTerra(mkinfo *v4.MarketingInfo) *schema.Set {
 	if mkinfo == nil {
 		return nil
 	}
-	mkinfos := []*v4.MarketingInfo{mkinfo}
-	mappedMkInfos := make([]interface{}, 0)
-	for _, mkinfo := range mkinfos {
-		mappedMkInfo := make(map[string]interface{})
-		mappedMkInfo["logo"] = mkinfo.Logo
-		mappedMkInfo["promotion"] = mkinfo.Promotion
-		processStepl := processStepFabricSpToTerra(mkinfo.ProcessSteps)
-		if processStepl != nil {
-			mappedMkInfo["process_step"] = processStepl
-		}
-		mappedMkInfos = append(mappedMkInfos, mappedMkInfo)
+	mappedMkInfo := make(map[string]interface{})
+	mappedMkInfo["logo"] = mkinfo.Logo
+	mappedMkInfo["promotion"] = mkinfo.Promotion
+	processStepl := processStepFabricSpToTerra(mkinfo.ProcessSteps)
+	if processStepl != nil {
+		mappedMkInfo["process_step"] = processStepl
 	}
 	marketingInfoSet := schema.NewSet(
 		schema.HashResource(&schema.Resource{Schema: createMarketingInfoSch()}),
-		mappedMkInfos,
+		[]interface{}{mappedMkInfo},
 	)
 	return marketingInfoSet
 }
 
-func accessPointColoFabricSpToTerra(accessPointColol []v4.ServiceProfileAccessPointColo) []map[string]interface{} {
+func accessPointColoFabricSpToTerra(accessPointColol []v4.ServiceProfileAccessPointColo) []interface{} {
 	if accessPointColol == nil {
 		return nil
 	}
-	mappedAccessPointColol := make([]map[string]interface{}, len(accessPointColol))
+	mappedAccessPointColol := make([]interface{}, len(accessPointColol))
```

Comment:
The `l` suffix to represent list is erroneous and confusing. We can just add an `s` to indicate the plurality of it.